Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+                            }
+                        }
+                    }
+                    elseif ($armResource.properties.auth.type.ToLower() -eq 'basic') {
+                        # username
+                        $hasUsername = [bool]($armResource.properties.auth.PSobject.Properties.name -match "username")
+                        if ($hasUsername) {
+                            $usernameProperty = $armResource.properties.auth.username
+                            $placeHoldersMatched = $usernameProperty | Select-String $placeHolderPatternMatches -AllMatches
+    
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.auth.username = "[[parameters('username')]"
+    
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'username' -isSecret $false
+                            }
+                        }
+    
+                        # password 
+                        $hasPassword = [bool]($armResource.properties.auth.PSobject.Properties.name -match "password")
+                        if ($hasPassword) {
+                            $passwordProperty = $armResource.properties.auth.password
+                            $placeHoldersMatched = $passwordProperty | Select-String $placeHolderPatternMatches -AllMatches
+    
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.auth.password = "[[parameters('password')]"
+    
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'password' -isSecret $true
+                            }
+                        }
+                    }
+                    elseif ($armResource.properties.auth.type.ToLower() -eq 'apikey') {
+                        # ApiKey 
+                        $hasApiKey = [bool]($armResource.properties.auth.PSobject.Properties.name -match "ApiKey")
+                        if ($hasApiKey) {
+                            $apiKeyProperty = $armResource.properties.auth.ApiKey
+                            $placeHoldersMatched = $apiKeyProperty | Select-String $placeHolderPatternMatches -AllMatches
+    
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
```

Comment:
check for this usage since it is unused most of the times

Reply:
updated